runtime/resolvers/legacy_metrics.go

```diff
@@ -0,0 +1,329 @@
+package resolvers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/mitchellh/hashstructure/v2"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/pkg/formatter"
+	"github.com/rilldata/rill/runtime/pkg/mapstructureutil"
+	"github.com/rilldata/rill/runtime/queries"
+	"go.uber.org/zap"
+)
+
+func init() {
+	runtime.RegisterResolverInitializer("legacy_metrics", newLegacyMetrics)
+}
+
+type legacyMetricsResolver struct {
+	runtime         *runtime.Runtime
+	instanceID      string
+	query           *runtime.Query
+	args            *legacyMetricsResolverArgs
+	metricsViewName string
+	rowCount        int
+	isFormatted     bool
+	logger          *zap.Logger
+}
+
+type legacyMetricsResolverProps struct {
+	QueryName     string `mapstructure:"query_name"`
+	QueryArgsJSON string `mapstructure:"query_args_json"`
+	RowCount      int    `mapstructure:"row_count"`
```

Comment:
Maybe call this `Limit`? I also wonder if it makes sense to move this to `ResolverOptions` and implementing it generically for all resolvers? Since for any resolver, the alert only needs the first row. But maybe that is over engineering.